tooling-cpu-load/src/test/java/org/jboss/eap/qe/microprofile/tooling/cpu/load/HighCPUUtilsTest.java

```diff
@@ -0,0 +1,70 @@
+package org.jboss.eap.qe.microprofile.tooling.cpu.load;
+
+import java.time.Duration;
+
+import javax.management.MBeanServerConnection;
+import javax.management.ObjectName;
+import javax.management.remote.JMXConnector;
+import javax.management.remote.JMXConnectorFactory;
+import javax.management.remote.JMXServiceURL;
+
+import org.awaitility.Awaitility;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.eap.qe.microprofile.tooling.cpu.load.utils.ProcessUtils;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.arquillian.ArquillianContainerProperties;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.arquillian.ArquillianDescriptorWrapper;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.creaper.ManagementClientProvider;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+
+/**
+ * Tests tooling for HighCPUUtils. Test CPU on the system is exhausted.
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class HighCPUUtilsTest {
+
+    private int cores = Runtime.getRuntime().availableProcessors();
+    private int coresUnderLoad = cores > 1 ? cores - 1 : 1;
+
+    @Test
+    public void testCauseMaximumCPULoadOnProcess() throws Exception {
+        Assert.assertTrue("This test must be run on RHEL or latest Fedora machines",
+                System.getProperty("os.name").contains("Linux") && (System.getProperty("os.version").contains("el")
+                        || System.getProperty("os.version").contains("fc2")));
+
+        Process cpuLoadProcess = null;
+        try (OnlineManagementClient client = ManagementClientProvider.onlineStandalone()) {
+            HighCPUUtils.causeMaximumCPULoadOnContainer(ProcessUtils.getProcessId(client),
+                    coresUnderLoad == 1 ? "0" : "0-" + coresUnderLoad, Duration.ofSeconds(10));
+
+            ArquillianContainerProperties props = new ArquillianContainerProperties(
+                    ArquillianDescriptorWrapper.getArquillianDescriptor());
+            Awaitility.await("CPU load generator does not work and is not causing any CPU load.")
```

Comment:
Is there other usecase for Awaitility library? If no, this can be replaced by simple cycle.

Reply:
Not in this code but I'm quite heavily using it in microprofile fault tolerance tests to test @Asynchronous MP FT annotation, CPU load or DB crash tests. I've found 20+ occurrences of it. This library is recommended to test asynchronous code in https://phauer.com/2019/modern-best-practices-testing-java/#use-awaitility-for-asserting-asynchronous-code

Also it's really tiny:
```
104K	tooling-cpu-load/target/dependency/awaitility-3.1.6.jar
```

Reply:
ok then